chat_stream.go

```diff
@@ -86,6 +86,13 @@ func (c *Client) CreateChatCompletionStream(
 	ctx context.Context,
 	request ChatCompletionRequest,
 ) (stream *ChatCompletionStream, err error) {
+	switch request.Model {
+	case GPT3Dot5Turbo0301, GPT3Dot5Turbo, GPT4, GPT40314, GPT432K0314, GPT432K:
```

Comment:
Are we sure this endpoint supports the `GPT4` model?

Reply:
Yes, as stated in the OpenAI documentation available at https://platform.openai.com/docs/models/model-endpoint-
  compatibility

Reply:
Thank you for the link, it's really helpful! I wonder if we can have this model-checking in a single place, so each method can just have something like:

```go
endpoint := "/v1/completions"
err = checkEndpointSupportsModel(endpoint, request.Model)
if err != nil {
    return
}
```

Reply:
This way we'll be able to generate proper error messages, as per your comment below